"use strict";(self.webpackChunkdocs=self.webpackChunkdocs||[]).push([[7676],{3905:(e,t,n)=>{n.d(t,{Zo:()=>p,kt:()=>g});var a=n(7294);function r(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function i(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?i(Object(n),!0).forEach((function(t){r(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):i(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function l(e,t){if(null==e)return{};var n,a,r=function(e,t){if(null==e)return{};var n,a,r={},i=Object.keys(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(a=0;a<i.length;a++)n=i[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var s=a.createContext({}),d=function(e){var t=a.useContext(s),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},p=function(e){var t=d(e.components);return a.createElement(s.Provider,{value:t},e.children)},m="mdxType",c={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},u=a.forwardRef((function(e,t){var n=e.components,r=e.mdxType,i=e.originalType,s=e.parentName,p=l(e,["components","mdxType","originalType","parentName"]),m=d(n),u=r,g=m["".concat(s,".").concat(u)]||m[u]||c[u]||i;return n?a.createElement(g,o(o({ref:t},p),{},{components:n})):a.createElement(g,o({ref:t},p))}));function g(e,t){var n=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=n.length,o=new Array(i);o[0]=u;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l[m]="string"==typeof e?e:r,o[1]=l;for(var d=2;d<i;d++)o[d]=n[d];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},7421:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>s,contentTitle:()=>o,default:()=>c,frontMatter:()=>i,metadata:()=>l,toc:()=>d});var a=n(7462),r=(n(7294),n(3905));const i={title:"Through Data Config",sidebar_position:2},o="Data Configuration",l={unversionedId:"loading-data/through-data-config",id:"loading-data/through-data-config",title:"Through Data Config",description:"You can load a large number of samples through data configurations. You need to (1) make a data config file (.json) that contains the information for individual samples, (2) store the config file in a HTTPS file server (e.g., AWS S3 or GitHub Gist), and (3) use it with the external parameter of the Chromoscope URL:",source:"@site/docs/loading-data/through-data-config.md",sourceDirName:"loading-data",slug:"/loading-data/through-data-config",permalink:"/loading-data/through-data-config",draft:!1,editUrl:"https://github.com/hms-dbmi/chromoscope/tree/master/docs/docs/loading-data/through-data-config.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{title:"Through Data Config",sidebar_position:2},sidebar:"docsSidebar",previous:{title:"Through Interface",permalink:"/loading-data/through-interface"},next:{title:"Data Formats",permalink:"/loading-data/data-formats"}},s={},d=[],p={toc:d},m="wrapper";function c(e){let{components:t,...n}=e;return(0,r.kt)(m,(0,a.Z)({},p,n,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"data-configuration"},"Data Configuration"),(0,r.kt)("p",null,"You can load a large number of samples through data configurations. You need to (1) make a data config file (",(0,r.kt)("inlineCode",{parentName:"p"},".json"),") that contains the information for individual samples, (2) store the config file in a HTTPS file server (e.g., AWS S3 or ",(0,r.kt)("a",{parentName:"p",href:"https://gist.github.com/"},"GitHub Gist"),"), and (3) use it with ",(0,r.kt)("a",{parentName:"p",href:"./url-parameters"},"the ",(0,r.kt)("inlineCode",{parentName:"a"},"external")," parameter of the Chromoscope URL"),":"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"// format\nhttps://chromoscope.bio/app/?external=[URL_TO_YOUR_CONFIG_FILE]\n\n// example\nhttps://chromoscope.bio/app/?external=https://gist.githubusercontent.com/sehilyi/a9bbbc3e63806d2282e1959e27a65a53/raw/b6c0ab07a220027196746f46607e916bd9751c70/goscan-multiple-samples.json\n")),(0,r.kt)("p",null,"For each sample, you need to prepare the following information in a JSON object."),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Property"),(0,r.kt)("th",{parentName:"tr",align:null},"Type"),(0,r.kt)("th",{parentName:"tr",align:null},"Note"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"id")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Unique ID.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"cancer")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Type of a cancer.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"assembly")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"'hg38'")," or ",(0,r.kt)("inlineCode",{parentName:"td"},"'hg19'")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Assembly.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"sv")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. An URL of the SV bedpe file (",(0,r.kt)("inlineCode",{parentName:"td"},".bedpe"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"cnv")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the CNV text file (",(0,r.kt)("inlineCode",{parentName:"td"},".tsv"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"drivers")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of a file that contains drivers (",(0,r.kt)("inlineCode",{parentName:"td"},".tsv")," or ",(0,r.kt)("inlineCode",{parentName:"td"},".json"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcf")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the point mutation file (",(0,r.kt)("inlineCode",{parentName:"td"},".vcf"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcfIndex")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the point mutation index file (",(0,r.kt)("inlineCode",{parentName:"td"},".tbi"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcf2")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the the indel file (",(0,r.kt)("inlineCode",{parentName:"td"},".vcf"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcf2Index")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the indel index file (",(0,r.kt)("inlineCode",{parentName:"td"},".tbi"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"bam")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the BAM file (",(0,r.kt)("inlineCode",{parentName:"td"},".bam"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"bai")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the BAM index file (",(0,r.kt)("inlineCode",{parentName:"td"},".bai"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"note")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. A textual annotation.")))),(0,r.kt)("p",null,"A single-sample example:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-js"},'{\n    "id": "7a921087-8e62-4a93-a757-fd8cdbe1eb8f",\n    "cancer": "ovarian",\n    "assembly": "hg19",\n    "sv": "https://s3.amazonaws.com/gosling-lang.org/data/SV/7a921087-8e62-4a93-a757-fd8cdbe1eb8f.pcawg_consensus_1.6.161022.somatic.sv.bedpe",\n    "cnv": "https://s3.amazonaws.com/gosling-lang.org/data/SV/7a921087-8e62-4a93-a757-fd8cdbe1eb8f.consensus.20170119.somatic.cna.annotated.txt"\n}\n')),(0,r.kt)("p",null,"A multi-sample example:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-js"},'[\n    {\n        "id": "SRR7890905",\n        "cancer": "breast",\n        "assembly": "hg38",\n        "drivers": "https://gist.githubusercontent.com/sehilyi/350b9e633c52ad97df00a0fc13a8839a/raw/c47b9ba33f1c9e187c69d1dadd01838db44d3b29/driver.txt",\n        "sv": "https://somatic-browser-test.s3.amazonaws.com/SVTYPE_SV_test_tumor_normal_with_panel.bedpe",\n        "cnv": "https://gist.githubusercontent.com/sehilyi/6fbceae35756b13472332d6b81b10803/raw/596428a8b0ebc00e7f8cbc52b050db0fbd6e19a5/SRR7890943.ascat.v3.cnv.tsv",\n        "bam": "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam",\n        "bai": "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam.bai",\n        "note": "This is a test note"\n    },\n    {\n        "id": "bc0dee07-de20-44d6-be65-05af7e63ac96",\n        "cancer": "gastric",\n        "assembly": "hg19",\n        "sv": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.pcawg_consensus_1.6.161116.somatic.sv.bedpe",\n        "cnv": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20170119.somatic.cna.txt",\n        "vcf": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20160830.somatic.snv_mnv.sorted.vcf.gz",\n        "vcfIndex": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20160830.somatic.snv_mnv.sorted.vcf.gz.tbi",\n        "vcf2": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20161006.somatic.indel.sorted.vcf.gz",\n        "vcf2Index": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20161006.somatic.indel.sorted.vcf.gz.tbi",\n        "note": "This is a test note"\n    }\n]\n')),(0,r.kt)("h1",{id:"genome-interpretation-panel"},"Genome Interpretation Panel"),(0,r.kt)("p",null,"The data for the genome interpretation panel can be recorded as additional features of the configuration file.\n",(0,r.kt)("img",{parentName:"p",src:"https://github.com/user-attachments/assets/02d15e04-0986-4e8e-8da8-2a07eb1119a4",alt:"Screenshot 2025-01-15 at 10 46 28\u202fAM"}),"\nFor each sample, there is an optional field: ",(0,r.kt)("inlineCode",{parentName:"p"},"clinicalInfo"),", which will display the clinical info if present.\nThe ",(0,r.kt)("inlineCode",{parentName:"p"},"clinicalInfo")," has the following fields:\n",(0,r.kt)("inlineCode",{parentName:"p"},"summary"),", ",(0,r.kt)("inlineCode",{parentName:"p"},"variants"),", ",(0,r.kt)("inlineCode",{parentName:"p"},"signatures")),(0,r.kt)("p",null,"The ",(0,r.kt)("inlineCode",{parentName:"p"},"summary")," field is a list of elements with two fields: ",(0,r.kt)("inlineCode",{parentName:"p"},"label")," and ",(0,r.kt)("inlineCode",{parentName:"p"},"value"),". Label can be any string, and will be displayed next to ",(0,r.kt)("inlineCode",{parentName:"p"},"value")," which can also be any string."),(0,r.kt)("p",null,"The ",(0,r.kt)("inlineCode",{parentName:"p"},"variants")," field is a list of variants, each with the following fields:\n| Property | Type | Note |\n|---|---|---|\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"gene")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"string")," | Required. Gene name |\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"chr")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"string")," | Required. Chromosome with the variant, including the chr prefix|\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"position")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"int")," | Required. Chromosomal coordinate of the mutation|\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"type")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"string")," | Optional. Type of variant, eg. bi-allelic, germline, deletion |\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"cDNA")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"string")," | Optional. cDNA coordinate of the mutation, eg. c.524G>A |\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"protein change")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"string")," | Optional. protein consequence of the mutation, eg. p.Arg175His |\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"VAF")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"float")," | Optional. Variant allele fraction of a mutation in the sample, eg. 0.45 |\n| ",(0,r.kt)("inlineCode",{parentName:"p"},"mutation")," | ",(0,r.kt)("inlineCode",{parentName:"p"},"string")," | Optional. Reference and alternative allele in genomic DNA, eg. G>T |"),(0,r.kt)("p",null,"Example of a configuration with the information necessary for the genome interpretation panel"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-js"},'[\n    {\n        id: "SRR7890905_Hartwig",\n        cancer: "breast",\n        assembly: "hg38",\n        vcf: "https://somatic-browser-test.s3.amazonaws.com/SNV_test_tumor_normal_with_panel.vcf.gz",\n        vcfIndex: "https://somatic-browser-test.s3.amazonaws.com/SNV_test_tumor_normal_with_panel.vcf.gz.tbi",\n        vcf2: "https://somatic-browser-test.s3.amazonaws.com/INDEL_test_tumor_normal_with_panel.vcf.gz",\n        vcf2Index: "https://somatic-browser-test.s3.amazonaws.com/INDEL_test_tumor_normal_with_panel.vcf.gz.tbi",\n        sv: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905/SRR7890905.gripss.filtered.bedpe",\n        cnv: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905/SRR7890905.purple.cnv.somatic.reformatted.tsv",\n        bam: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam",\n        bai: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam.bai",\n        note: "CNV profile - Purple. SVs - Gridss. Mutations and indels - Sentieon",\n    clinicalInfo: {\n    summary: [\n        {\n            label: "grade",\n            value: "2"\n        },\n        {\n            label: "age",\n            value: "32"\n        },\n        {\n            label: "AIMS Subtype",\n            value: "Basal"\n        },\n        {\n            label: "Treatment",\n            value: "None"\n        },\n        {\n            label: "Lymph node status",\n            value: "Negative"\n        }\n        ],\n        variants: [\n        {\n            gene: "TP53",\n            type: "biallelic",\n            cDNA: "c.524G>A",\n            protein change: "p.Arg175His",\n            VAF: "0.97",\n            chr: "chr17",\n            position: "7677976",\n            mutation: "C>T"\n        },\n        {\n            gene: "BRCA2",\n            cDNA: "c.4777G>T",\n            protein change: "p.Glu1593Ter",\n            VAF: "0.45",\n            chr: "chr13",\n            position: "32357888",\n            mutation: "G>T"\n        },\n        {\n            gene: "PTEN",\n            type: "deletion",\n            chr: "chr10",\n            position: "87917777"\n        },\n        {\n            gene: "CDKN2A",\n            type: "deletion",\n            chr: "chr9",\n            position: "21981538"\n        },\n        {\n            gene: "MET",\n            type: "amplification",\n            chr: "chr7",\n            position: "116735286"\n        }\n        ],\n        signatures: [\n        {\n            type: "point_mutations",\n            count: "5100",\n            label: "HRD-attributed point mutations",\n            hrDetect: true\n        },\n        {\n            type: "indels",\n            count: "500",\n            label: "HRD-attributed small deletions with micro-homology",\n            hrDetect: true\n        },\n        {\n            type: "svs_duplications",\n            count: "120",\n            label: "HRD-attributed small tandem duplications",\n            hrDetect: true\n        },\n        {\n            type: "svs_deletions",\n            count: "50",\n            label: "HRD-attributed small deletions",\n            hrDetect: true\n        },\n        {\n            type: "point_mutations",\n            count: "1000",\n            label: "APOBEC-attributed small substitutions",\n            hrDetect: false\n        }\n        ]\n        }\n    }\n]\n')))}c.isMDXComponent=!0}}]);