"use strict";(self.webpackChunkdocs=self.webpackChunkdocs||[]).push([[7676],{3905:(t,e,n)=>{n.d(e,{Zo:()=>p,kt:()=>g});var a=n(7294);function r(t,e,n){return e in t?Object.defineProperty(t,e,{value:n,enumerable:!0,configurable:!0,writable:!0}):t[e]=n,t}function i(t,e){var n=Object.keys(t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(t);e&&(a=a.filter((function(e){return Object.getOwnPropertyDescriptor(t,e).enumerable}))),n.push.apply(n,a)}return n}function l(t){for(var e=1;e<arguments.length;e++){var n=null!=arguments[e]?arguments[e]:{};e%2?i(Object(n),!0).forEach((function(e){r(t,e,n[e])})):Object.getOwnPropertyDescriptors?Object.defineProperties(t,Object.getOwnPropertyDescriptors(n)):i(Object(n)).forEach((function(e){Object.defineProperty(t,e,Object.getOwnPropertyDescriptor(n,e))}))}return t}function o(t,e){if(null==t)return{};var n,a,r=function(t,e){if(null==t)return{};var n,a,r={},i=Object.keys(t);for(a=0;a<i.length;a++)n=i[a],e.indexOf(n)>=0||(r[n]=t[n]);return r}(t,e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(t);for(a=0;a<i.length;a++)n=i[a],e.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(t,n)&&(r[n]=t[n])}return r}var d=a.createContext({}),s=function(t){var e=a.useContext(d),n=e;return t&&(n="function"==typeof t?t(e):l(l({},e),t)),n},p=function(t){var e=s(t.components);return a.createElement(d.Provider,{value:e},t.children)},m="mdxType",c={inlineCode:"code",wrapper:function(t){var e=t.children;return a.createElement(a.Fragment,{},e)}},u=a.forwardRef((function(t,e){var n=t.components,r=t.mdxType,i=t.originalType,d=t.parentName,p=o(t,["components","mdxType","originalType","parentName"]),m=s(n),u=r,g=m["".concat(d,".").concat(u)]||m[u]||c[u]||i;return n?a.createElement(g,l(l({ref:e},p),{},{components:n})):a.createElement(g,l({ref:e},p))}));function g(t,e){var n=arguments,r=e&&e.mdxType;if("string"==typeof t||r){var i=n.length,l=new Array(i);l[0]=u;var o={};for(var d in e)hasOwnProperty.call(e,d)&&(o[d]=e[d]);o.originalType=t,o[m]="string"==typeof t?t:r,l[1]=o;for(var s=2;s<i;s++)l[s]=n[s];return a.createElement.apply(null,l)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},7421:(t,e,n)=>{n.r(e),n.d(e,{assets:()=>d,contentTitle:()=>l,default:()=>c,frontMatter:()=>i,metadata:()=>o,toc:()=>s});var a=n(7462),r=(n(7294),n(3905));const i={title:"Through Data Config",sidebar_position:2},l="Data Configuration",o={unversionedId:"loading-data/through-data-config",id:"loading-data/through-data-config",title:"Through Data Config",description:"You can load a large number of samples through data configurations. You need to (1) make a data config file (.json) that contains the information for individual samples, (2) store the config file in a HTTPS file server (e.g., AWS S3 or GitHub Gist), and (3) use it with the external parameter of the Chromoscope URL:",source:"@site/docs/loading-data/through-data-config.md",sourceDirName:"loading-data",slug:"/loading-data/through-data-config",permalink:"/loading-data/through-data-config",draft:!1,editUrl:"https://github.com/hms-dbmi/chromoscope/tree/master/docs/docs/loading-data/through-data-config.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{title:"Through Data Config",sidebar_position:2},sidebar:"docsSidebar",previous:{title:"Through Interface",permalink:"/loading-data/through-interface"},next:{title:"Data Formats",permalink:"/loading-data/data-formats"}},d={},s=[{value:"Notes",id:"notes",level:2}],p={toc:s},m="wrapper";function c(t){let{components:e,...n}=t;return(0,r.kt)(m,(0,a.Z)({},p,n,{components:e,mdxType:"MDXLayout"}),(0,r.kt)("h1",{id:"data-configuration"},"Data Configuration"),(0,r.kt)("p",null,"You can load a large number of samples through data configurations. You need to (1) make a data config file (",(0,r.kt)("inlineCode",{parentName:"p"},".json"),") that contains the information for individual samples, (2) store the config file in a HTTPS file server (e.g., AWS S3 or ",(0,r.kt)("a",{parentName:"p",href:"https://gist.github.com/"},"GitHub Gist"),"), and (3) use it with ",(0,r.kt)("a",{parentName:"p",href:"./url-parameters"},"the ",(0,r.kt)("inlineCode",{parentName:"a"},"external")," parameter of the Chromoscope URL"),":"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre"},"// format\nhttps://chromoscope.bio/app/?external=[URL_TO_YOUR_CONFIG_FILE]\n\n// example\nhttps://chromoscope.bio/app/?external=https://gist.githubusercontent.com/sehilyi/a9bbbc3e63806d2282e1959e27a65a53/raw/b6c0ab07a220027196746f46607e916bd9751c70/goscan-multiple-samples.json\n")),(0,r.kt)("p",null,"For each sample, you need to prepare the following information in a JSON object."),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Property"),(0,r.kt)("th",{parentName:"tr",align:null},"Type"),(0,r.kt)("th",{parentName:"tr",align:null},"Note"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"id")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Unique ID.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"cancer")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Type of a cancer.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"assembly")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"'hg38'")," or ",(0,r.kt)("inlineCode",{parentName:"td"},"'hg19'")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Assembly.")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"sv")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. An URL of the SV bedpe file (",(0,r.kt)("inlineCode",{parentName:"td"},".bedpe"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"cnv")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the CNV text file (",(0,r.kt)("inlineCode",{parentName:"td"},".tsv"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"drivers")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of a file that contains drivers (",(0,r.kt)("inlineCode",{parentName:"td"},".tsv")," or ",(0,r.kt)("inlineCode",{parentName:"td"},".json"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcf")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the point mutation file (",(0,r.kt)("inlineCode",{parentName:"td"},".vcf"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcfIndex")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the point mutation index file (",(0,r.kt)("inlineCode",{parentName:"td"},".tbi"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcf2")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the the indel file (",(0,r.kt)("inlineCode",{parentName:"td"},".vcf"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"vcf2Index")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the indel index file (",(0,r.kt)("inlineCode",{parentName:"td"},".tbi"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"bam")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the BAM file (",(0,r.kt)("inlineCode",{parentName:"td"},".bam"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"bai")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. An URL of the BAM index file (",(0,r.kt)("inlineCode",{parentName:"td"},".bai"),").")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"note")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. A textual annotation.")))),(0,r.kt)("p",null,"A single-sample example:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-js"},'{\n    "id": "7a921087-8e62-4a93-a757-fd8cdbe1eb8f",\n    "cancer": "ovarian",\n    "assembly": "hg19",\n    "sv": "https://s3.amazonaws.com/gosling-lang.org/data/SV/7a921087-8e62-4a93-a757-fd8cdbe1eb8f.pcawg_consensus_1.6.161022.somatic.sv.bedpe",\n    "cnv": "https://s3.amazonaws.com/gosling-lang.org/data/SV/7a921087-8e62-4a93-a757-fd8cdbe1eb8f.consensus.20170119.somatic.cna.annotated.txt"\n}\n')),(0,r.kt)("p",null,"A multi-sample example:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-js"},'[\n    {\n        "id": "SRR7890905",\n        "cancer": "breast",\n        "assembly": "hg38",\n        "drivers": "https://gist.githubusercontent.com/sehilyi/350b9e633c52ad97df00a0fc13a8839a/raw/c47b9ba33f1c9e187c69d1dadd01838db44d3b29/driver.txt",\n        "sv": "https://somatic-browser-test.s3.amazonaws.com/SVTYPE_SV_test_tumor_normal_with_panel.bedpe",\n        "cnv": "https://gist.githubusercontent.com/sehilyi/6fbceae35756b13472332d6b81b10803/raw/596428a8b0ebc00e7f8cbc52b050db0fbd6e19a5/SRR7890943.ascat.v3.cnv.tsv",\n        "bam": "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam",\n        "bai": "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam.bai",\n        "note": "This is a test note"\n    },\n    {\n        "id": "bc0dee07-de20-44d6-be65-05af7e63ac96",\n        "cancer": "gastric",\n        "assembly": "hg19",\n        "sv": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.pcawg_consensus_1.6.161116.somatic.sv.bedpe",\n        "cnv": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20170119.somatic.cna.txt",\n        "vcf": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20160830.somatic.snv_mnv.sorted.vcf.gz",\n        "vcfIndex": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20160830.somatic.snv_mnv.sorted.vcf.gz.tbi",\n        "vcf2": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20161006.somatic.indel.sorted.vcf.gz",\n        "vcf2Index": "https://somatic-browser-test.s3.amazonaws.com/cdk12cancers/bc0dee07-de20-44d6-be65-05af7e63ac96.consensus.20161006.somatic.indel.sorted.vcf.gz.tbi",\n        "note": "This is a test note"\n    }\n]\n')),(0,r.kt)("h1",{id:"genome-interpretation-panel"},"Genome Interpretation Panel"),(0,r.kt)("p",null,"The data for the genome interpretation panel can be recorded as additional features of the configuration file.\n",(0,r.kt)("img",{parentName:"p",src:"https://github.com/user-attachments/assets/02d15e04-0986-4e8e-8da8-2a07eb1119a4",alt:"Screenshot 2025-01-15 at 10 46 28\u202fAM"}),"\nFor each sample, there is an optional property (",(0,r.kt)("inlineCode",{parentName:"p"},"clinicalInfo"),") to provide clinical information. If present, this information will be displayed in Chromoscope (the panel on the right-most side)."),(0,r.kt)("p",null,"You can provide three fields under the ",(0,r.kt)("inlineCode",{parentName:"p"},"clinicalInfo")," property: ",(0,r.kt)("inlineCode",{parentName:"p"},"summary"),", ",(0,r.kt)("inlineCode",{parentName:"p"},"variants"),", ",(0,r.kt)("inlineCode",{parentName:"p"},"signatures")),(0,r.kt)("p",null,"The ",(0,r.kt)("inlineCode",{parentName:"p"},"summary")," field is a list of elements with two fields: ",(0,r.kt)("inlineCode",{parentName:"p"},"label")," and ",(0,r.kt)("inlineCode",{parentName:"p"},"value"),". Label can be any string, and will be displayed next to ",(0,r.kt)("inlineCode",{parentName:"p"},"value")," which can also be any string."),(0,r.kt)("p",null,"The ",(0,r.kt)("inlineCode",{parentName:"p"},"variants")," field is a list of variants, each with the following fields:"),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},"Property"),(0,r.kt)("th",{parentName:"tr",align:null},"Type"),(0,r.kt)("th",{parentName:"tr",align:null},"Note"))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"gene")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Gene name")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"chr")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Chromosome with the variant, including the chr prefix")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"position")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"number")),(0,r.kt)("td",{parentName:"tr",align:null},"Required. Chromosomal coordinate of the mutation")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"type")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. Type of variant, e.g., ",(0,r.kt)("inlineCode",{parentName:"td"},'"bi-allelic"'),", ",(0,r.kt)("inlineCode",{parentName:"td"},'"germline"'),", ",(0,r.kt)("inlineCode",{parentName:"td"},'"deletion"'))),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"cDNA")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. cDNA coordinate of the mutation, e.g., ",(0,r.kt)("inlineCode",{parentName:"td"},'"c.524G>A"'))),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"protein change")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. protein consequence of the mutation, e.g., ",(0,r.kt)("inlineCode",{parentName:"td"},'"p.Arg175His"'))),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"VAF")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"number")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. Variant allele fraction of a mutation in the sample, e.g., ",(0,r.kt)("inlineCode",{parentName:"td"},"0.45"))),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"mutation")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. Reference and alternative allele in genomic DNA, e.g., ",(0,r.kt)("inlineCode",{parentName:"td"},'"G>T"'))),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"sv_id")),(0,r.kt)("td",{parentName:"tr",align:null},(0,r.kt)("inlineCode",{parentName:"td"},"string")),(0,r.kt)("td",{parentName:"tr",align:null},"Optional. If the variant is a structural variant, if the ",(0,r.kt)("inlineCode",{parentName:"td"},"sv_id")," is provided, the variant will be highlighted, e.g., ",(0,r.kt)("inlineCode",{parentName:"td"},'"bnd_1012513"'))))),(0,r.kt)("p",null,"}"),(0,r.kt)("p",null,"Refer to an example configuration file that contains the clinical information necessary for displaying the genome interpretation panel:"),(0,r.kt)("pre",null,(0,r.kt)("code",{parentName:"pre",className:"language-js"},'{\n    id: "SRR7890905_Hartwig",\n    cancer: "breast",\n    assembly: "hg38",\n    vcf: "https://somatic-browser-test.s3.amazonaws.com/SNV_test_tumor_normal_with_panel.vcf.gz",\n    vcfIndex: "https://somatic-browser-test.s3.amazonaws.com/SNV_test_tumor_normal_with_panel.vcf.gz.tbi",\n    vcf2: "https://somatic-browser-test.s3.amazonaws.com/INDEL_test_tumor_normal_with_panel.vcf.gz",\n    vcf2Index: "https://somatic-browser-test.s3.amazonaws.com/INDEL_test_tumor_normal_with_panel.vcf.gz.tbi",\n    sv: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905/SRR7890905.gripss.filtered.bedpe",\n    cnv: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905/SRR7890905.purple.cnv.somatic.reformatted.tsv",\n    bam: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam",\n    bai: "https://somatic-browser-test.s3.amazonaws.com/SRR7890905_GAPFI2USVS21.bam.bai",\n    note: "CNV profile - Purple. SVs - Gridss. Mutations and indels - Sentieon",\n    clinicalInfo: {\n        summary: [\n            {\n                label: "grade",\n                value: "2"\n            },\n            {\n                label: "age",\n                value: "32"\n            },\n            {\n                label: "AIMS Subtype",\n                value: "Basal"\n            },\n            {\n                label: "Treatment",\n                value: "None"\n            },\n            {\n                label: "Lymph node status",\n                value: "Negative"\n            }\n        ],\n        variants: [\n            {\n                gene: "TP53",\n                type: "biallelic",\n                cDNA: "c.524G>A",\n                protein change: "p.Arg175His",\n                VAF: "0.97",\n                chr: "chr17",\n                position: "7677976",\n                mutation: "C>T"\n            },\n            {\n                gene: "BRCA2",\n                cDNA: "c.4777G>T",\n                protein change: "p.Glu1593Ter",\n                VAF: "0.45",\n                chr: "chr13",\n                position: "32357888",\n                mutation: "G>T"\n            },\n            {\n                gene: "PTEN",\n                type: "deletion",\n                chr: "chr10",\n                position: "89637467",\n                sv_id: "bnd_1012513"\n            },\n            {\n                gene: "CDKN2A",\n                type: "deletion",\n                chr: "chr9",\n                position: "21981538"\n            },\n            {\n                gene: "MET",\n                type: "amplification",\n                chr: "chr7",\n                position: "116735286"\n            }\n        ],\n        signatures: [\n            {\n                type: "point_mutations",\n                count: "5100",\n                label: "HRD-attributed point mutations",\n                hrDetect: true\n            },\n            {\n                type: "indels",\n                count: "500",\n                label: "HRD-attributed small deletions with micro-homology",\n                hrDetect: true\n            },\n            {\n                type: "svs_duplications",\n                count: "120",\n                label: "HRD-attributed small tandem duplications",\n                hrDetect: true\n            },\n            {\n                type: "svs_deletions",\n                count: "50",\n                label: "HRD-attributed small deletions",\n                hrDetect: true\n            },\n            {\n                type: "point_mutations",\n                count: "1000",\n                label: "APOBEC-attributed small substitutions",\n                hrDetect: false\n            }\n        ]\n    }\n}\n')),(0,r.kt)("p",null,"Another example of a configuration file that includes information for the Genome Interpretation Panel is available here:  "),(0,r.kt)("p",null,(0,r.kt)("a",{parentName:"p",href:"https://somatic-browser-test.s3.us-east-1.amazonaws.com/SPECTRUM/SPECTRUM_config_with_clinicalInfo_sorted_aws.json"},"MSK SPECTRUM Configuration")),(0,r.kt)("hr",null),(0,r.kt)("h2",{id:"notes"},"Notes"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"HRDetect status encoding"),(0,r.kt)("br",{parentName:"p"}),"\n","The HRDetect status is determined by checking if any of the ",(0,r.kt)("inlineCode",{parentName:"p"},'"signature"')," items have ",(0,r.kt)("inlineCode",{parentName:"p"},"hrDetect = true"),". Otherwise, it is ",(0,r.kt)("inlineCode",{parentName:"p"},"false"),".  "),(0,r.kt)("ul",{parentName:"li"},(0,r.kt)("li",{parentName:"ul"},"If the ",(0,r.kt)("inlineCode",{parentName:"li"},"hrDetect")," field is not provided, the status defaults to ",(0,r.kt)("strong",{parentName:"li"},'"Negative"'),"."))),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},(0,r.kt)("strong",{parentName:"p"},"Variant highlighting mode"),(0,r.kt)("br",{parentName:"p"}),"\n","Variants are highlighted in the visualization if they are selected from the panel on the right.  "),(0,r.kt)("ul",{parentName:"li"},(0,r.kt)("li",{parentName:"ul"},"Implemented for ",(0,r.kt)("strong",{parentName:"li"},"point mutations")," and ",(0,r.kt)("strong",{parentName:"li"},"structural variants"),".  "),(0,r.kt)("li",{parentName:"ul"},"Not currently implemented for ",(0,r.kt)("strong",{parentName:"li"},"indels"),".  "),(0,r.kt)("li",{parentName:"ul"},"Matching rules:  ",(0,r.kt)("ul",{parentName:"li"},(0,r.kt)("li",{parentName:"ul"},"For point mutations \u2192 matched by chromosome name and chromosome coordinates.  "),(0,r.kt)("li",{parentName:"ul"},"For structural variants \u2192 matched by the ",(0,r.kt)("inlineCode",{parentName:"li"},"sv_id")," field, if present.")))))))}c.isMDXComponent=!0}}]);